src/psyclone/tests/utilities.py

```diff
@@ -109,7 +133,26 @@ def print_diffs(expected, actual):
 
 
 # =============================================================================
-class Compile(object):
+@contextmanager
+def change_dir(new_dir):
```

Comment:
I like this a lot. As noted in the tests, I think its usage there *might* be cleaner if we had a separate fixture that provided this functionality. Would you mind trying it and seeing whether it removes the need for the `with xxx:` in the tests?

Reply:
All done.